### PR TITLE
Waf active rules

### PR DESCRIPTION
**Pull request by oscarDovao**
This PR contains the waf active rules implementation on the provider. Active rules are implemented using TypeSet Terraform schema type and rules can be added and deleted in bulk so only 1 or 2 requests per single update is performed.
Batch updates have not been implemented. it depends how we deal with large amount of rules, we may need batch updates.

### CODE REVIEW

File: fastly/block_fastly_waf_configuration_v1_active_rule.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
+package fastly
+
+import (
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
```
Comment (phamann) [inline fastly/block_fastly_waf_configuration_v1_active_rule.go]: Please separate standard library imports from others.

File: fastly/block_fastly_waf_configuration_v1_active_rule.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+var activeRule = &schema.Schema{
+	Type:     schema.TypeSet,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"status": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The Web Application Firewall rule's status. Allowed values are (log, block and score)",
```
Comment (phamann) [inline fastly/block_fastly_waf_configuration_v1_active_rule.go]:
```suggestion
				Description:  "The Web Application Firewall rule's status. Allowed values are (log, block and score).",
```

File: fastly/block_fastly_waf_configuration_v1_active_rule.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+				ValidateFunc: validateRuleStatusType(),
+			},
+			"modsec_rule_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The Web Application Firewall rule's modsec id",
```
Comment (phamann) [inline fastly/block_fastly_waf_configuration_v1_active_rule.go]:
```suggestion
				Description: "The Web Application Firewall rule's modsec ID.",
```

File: fastly/block_fastly_waf_configuration_v1_active_rule.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+			},
+			"revision": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The Web Application Firewall rule's revision",
```
Comment (phamann) [inline fastly/block_fastly_waf_configuration_v1_active_rule.go]:
```suggestion
				Description: "The Web Application Firewall rule's revision.",
```

File: fastly/block_fastly_waf_configuration_v1_active_rule_test.go [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+import (
+	"fmt"
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"reflect"
+	"sort"
+	"testing"
```
Comment (phamann) [inline fastly/block_fastly_waf_configuration_v1_active_rule_test.go]: Please separate standard library imports from others.

File: fastly/block_fastly_waf_configuration_v1_active_rule.go (hunk 1 above)
Comment (oscarDovao) [reply fastly/block_fastly_waf_configuration_v1_active_rule.go]: imports organised

File: fastly/block_fastly_waf_configuration_v1_active_rule_test.go (hunk 5 above)
Comment (oscarDovao) [reply fastly/block_fastly_waf_configuration_v1_active_rule_test.go]: imports organised

**Pull request merged**